src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -315,13 +315,15 @@ class InstanceOpFactoryImpl(
     offer: Mesos.Offer,
     resourceMatch: ResourceMatcher.ResourceMatch): InstanceOp = {
 
-    val localVolumes: Seq[(DiskSource, Task.LocalVolume)] =
+    val localVolumes: Seq[(Option[String], DiskSource, Task.LocalVolume)] =
       resourceMatch.localVolumes.map {
-        case (source, volume, mount) =>
-          (source, Task.LocalVolume(Task.LocalVolumeId(runSpec.id, volume, mount), volume, mount))
+        case (providerId, source, volumeWithMount) =>
+          (providerId, source, Task.LocalVolume(
+            Task.LocalVolumeId(runSpec.id, volumeWithMount.volume, volumeWithMount.mount),
+            volumeWithMount.volume, volumeWithMount.mount))
       }
 
-    val persistentVolumeIds = localVolumes.map { case (_, localVolume) => localVolume.id }
+    val persistentVolumeIds = localVolumes.map { case (_, _, localVolume) => localVolume.id }
```

Comment:
if you go with case class even this will be much more readable from my point of view :)

Reply:
Done.